### DISCUSSION TITLE
Trapezoidal Unit FMM from Pyleecan

### DISCUSSION THREAD

**Discussion by bcaldeirar764** [Eomys/pyleecan#674] [Q&A]
Hello, everyone!

I'm still trying to extract the winding function curve (unit magnetomotive force) from Pyleecan. Through the graphical interface, the generated file results in an ideal rectangular wave, which is an approximation of the trapezoidal wave I'm looking for. 

Does anyone know how to obtain the trapezoidal wave that represents the winding function of the machine? (I am using the Toyota_Prius.json file as an example.) If I'm not mistaken, the command `stator.winding.comp_connection_mat()` (from the example code at https://www.pyleecan.org/01_tuto_Machine.html) obtains the winding pattern. Despite this, the command `print(wind_mat)` generates a matrix that results in an ideal rectangular wave. How can I obtain the trapezoidal curve I'm looking for? 

Thank you very much for the help!

**Comment by SebGue** [comment]
Hello @bcaldeirar764,

Have you found your answer meanwhile? Though I don't know exactly what are looking for, maybe you should have a look at [swat-em](https://swat-em.readthedocs.io/en/latest/). 

Best regards, Sebastian
